indexing-service/src/main/java/io/druid/indexing/overlord/ForkingTaskRunner.java

```diff
@@ -230,6 +232,24 @@ public TaskStatus call()
                                 }
                               }
 
+                              // Add dataSource and taskId for metrics
+                              command.add(
+                                  String.format(
+                                      "-D%s%s=%s",
```

Comment:
I tested this with having blank spaces and quotes inside datasource name and its working fine. 
don't think we need to do escaping there.